lib/qam.pm

```diff
@@ -84,9 +84,15 @@ sub add_test_repositories {
     # please be sure that the PATCH_TEST_REPO is empty.
     @repos = split(',', $oldrepo) if ($oldrepo);
 
-    for my $var (@repos) {
-        zypper_call("--no-gpg-checks ar -f $gpg -n 'TEST_$counter' $var 'TEST_$counter'");
-        $counter++;
+    if (get_var("NO_ADD_REPO")) {
+        # If we don't want to add again (and duplicate) repositories that were already added during install,
+        # we still need to disable gpg check for all repositories.
+        zypper_call('--gpg-auto-import-keys ref', timeout => 1400, exitcode => [0, 106]);
+    } else {
+        for my $var (@repos) {
+            zypper_call("--no-gpg-checks ar -f $gpg -n 'TEST_$counter' $var 'TEST_$counter'");
+            $counter++;
+        }
     }
 
     if (is_sle('=12-SP2')) {
```

Comment:
In the end, I modified the lib as we may need the piece of code below, this way we can apply this change in every maintained product, the oldest being sle12-sp2, as far as I could see.